### ISSUE TITLE
Task 8c: In-Situ Recording Evaluation Protocol

### ISSUE THREAD

**Issue by ToaruPen** [ToaruPen/Wooly-Fluffy#35]
# Task 8c: In-Situ Recording Evaluation Protocol

- Block: **H — Hardening**
- Source of truth: this issue (inlined from `.sisyphus/plans/wooly-fluffy.md`)

## Dependencies
- #12 (Task 8: Sound Detection)
- #28 (Task 8b: Audio HAL)
- #21 (Task 15: Voice Pipeline with silero-vad)

## Blocks
None (evaluation tool, no downstream dependencies)

---

## Specification

Implement a **recording and evaluation toolkit** for calibrating sound detection and VAD thresholds using real facility recordings:

### Recording tool (CLI command)
- `python -m daemon.tools.record_evaluation --duration 300 --output data/evaluation/`
- Records raw audio from Audio HAL for specified duration
- Saves as timestamped WAV files with metadata JSON (device type, location notes, noise conditions)
- Staff can annotate: "bell at 0:15", "multiple children talking at 1:30", "quiet period 2:00-2:30"
- No persistent storage of children's speech — recordings are for calibration only, deleted after evaluation

### Evaluation runner (CLI command)
- `python -m daemon.tools.evaluate_detection --recordings data/evaluation/ --config settings.yaml`
- Runs sound detector on recorded audio with current threshold settings
- Compares detections against staff annotations (ground truth)
- Outputs: precision, recall, F1 score, false positive rate, detection latency distribution
- Also runs VAD evaluation: speech segments detected vs annotated

### Threshold optimizer (CLI command)
- `python -m daemon.tools.optimize_thresholds --recordings data/evaluation/ --metric f1`
- Grid search over key parameters: `sensitivity_threshold`, `energy_gate_threshold`, `speech_prob_threshold`, `overlap_hold_ms`
- Outputs recommended settings.yaml values based on best metric
- Prints comparison table: current settings vs recommended

### Annotation format (JSON)
```json
{
  "recording_id": "2026-03-15_1430",
  "events": [
    {"time_s": 15.2, "type": "bell", "notes": "clean bell ring"},
    {"time_s": 90.5, "type": "multi_voice", "notes": "3 children talking"},
    {"time_s": 120.0, "end_s": 150.0, "type": "quiet", "notes": "homework time"}
  ]
}
```

### Integration with Staff UI (Task 23 will add UI for this)
- Recording start/stop via API
- Annotation via API (staff marks events while recording)
- Results displayed in Bell Tuning UI panel

### Must NOT Do
- Do not persist children's speech recordings beyond evaluation session (delete after analysis)
- Do not implement automatic labeling (annotations are manual by staff)
- Do not run evaluation on live audio (offline evaluation only)
- Do not optimize for metrics without staff review of recommended thresholds

---

## Files to Create
| File | Description |
|------|-------------|
| `daemon/src/daemon/tools/record_evaluation.py` | CLI recording tool (`python -m daemon.tools.record_evaluation`) |
| `daemon/src/daemon/tools/evaluate_detection.py` | CLI evaluation runner with precision/recall/F1 |
| `daemon/src/daemon/tools/optimize_thresholds.py` | CLI grid search threshold optimizer |
| `daemon/src/daemon/tools/annotation_schema.py` | Annotation JSON schema and validation |
| `tests/tools/test_record_evaluation.py` | Recording tool tests |
| `tests/tools/test_evaluate_detection.py` | Evaluation runner tests |
| `tests/tools/test_optimize_thresholds.py` | Threshold optimizer tests |
| `tests/tools/test_annotation_schema.py` | Annotation schema validation tests |
| `data/evaluation/.gitkeep` | Evaluation data directory (recordings are ephemeral) |

## Existing Codebase Conventions
- Python strict typing: `mypy --strict` + `pyright` must pass
- CLI tools as `python -m` modules under `daemon/src/daemon/tools/`
- Tests mirror source structure under `tests/`
- scikit-learn for metrics computation

---

## Acceptance Criteria
- [ ] TDD: `pytest tests/tools/` passes with ≥6 tests
- [ ] Recording tool: captures audio and saves WAV + metadata JSON
- [ ] Evaluation runner: computes precision/recall/F1 against annotated ground truth
- [ ] Threshold optimizer: grid search produces recommended settings with comparison table
- [ ] Annotation format: JSON schema validates staff annotations
- [ ] Privacy: recordings auto-deleted after evaluation completes (configurable retention)
- [ ] API endpoints for recording control (start/stop/annotate)

## QA Scenarios

```
Scenario: Record, annotate, and evaluate detection performance
  Tool: Bash (pytest)
  Preconditions: Sound detector initialized, mock audio HAL with pre-recorded facility audio
  Steps:
    1. Run recording tool: `python -m daemon.tools.record_evaluation --duration 10 --output /tmp/eval/`
    2. Assert WAV file created with correct sample rate and metadata JSON
    3. Create annotation file with 2 bell events and 1 multi_voice event
    4. Run evaluation: `python -m daemon.tools.evaluate_detection --recordings /tmp/eval/`
    5. Assert output contains precision, recall, F1 for sound detection
    6. Assert output contains VAD evaluation metrics
    7. Verify recordings deleted after evaluation (or after configurable retention)
  Expected Result: Complete record → annotate → evaluate pipeline with quantified metrics
  Failure Indicators: No metrics output, wrong file format, recordings not cleaned up
  Evidence: .sisyphus/evidence/task-8c-evaluation.txt

Scenario: Threshold optimization produces recommendations
  Tool: Bash (pytest)
  Preconditions: Evaluation recordings with annotations exist
  Steps:
    1. Run optimizer: `python -m daemon.tools.optimize_thresholds --recordings /tmp/eval/ --metric f1`
    2. Assert output contains comparison table (current vs recommended thresholds)
    3. Assert recommended thresholds improve F1 over current settings
    4. Assert output includes parameter names: sensitivity_threshold, energy_gate_threshold, speech_prob_threshold
  Expected Result: Data-driven threshold recommendations with clear improvement metrics
  Failure Indicators: No recommendations, worse metrics, missing parameters
  Evidence: .sisyphus/evidence/task-8c-optimization.txt
```

---

## TDD Workflow
1. Write failing tests for annotation schema validation
2. Implement annotation schema
3. Write failing tests for recording tool (mock audio HAL)
4. Implement recording tool
5. Write failing tests for evaluation runner (mock detections vs ground truth)
6. Implement evaluation runner with scikit-learn metrics
7. Write failing tests for threshold optimizer
8. Implement grid search optimizer
9. Verify all 6+ tests pass

## Quality Gates
```bash
uv run ruff format --check daemon tests scripts
uv run ruff check daemon tests scripts
uv run mypy --strict daemon/src tests
uv run pyright daemon/src
uv run pytest tests/tools/ -v
```

## ADR Requirement
No new ADR required.

## Commit Message
`feat(tools): in-situ recording evaluation protocol with annotation, metrics, and threshold optimizer`

## External References
- Oracle consultation (session `ses_322ead4a3ffeLX1xKcTVVDDmCD`): "必ず施設の実録音で評価セットを作ってください。ここをやらないと、どのモデル選択も机上最適化で終わります。" — This task exists because of this Oracle recommendation.
- scikit-learn metrics: `sklearn.metrics.precision_recall_fscore_support` — For computing detection evaluation metrics
- Sound detection calibration: Oracle recommended "校正サンプルは静かな近接録音だけでなく、実環境で取り直す前提にしてください"
